### PR TITLE
planner/core: always add projection to agg that is pushed to tiflash

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve?

Problem Summary: After implementing count-distinct agg totally push-down, sometimes agg's parent will be a non-projection, which results in unexpected error on TiFlash. 

### What is changed and how it works?

What's Changed:
We fix it by always adding a projection for scalar agg that is pushed down.

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- planner/core: always add projection to agg that is pushed to tiflash

### CODE REVIEW

File: planner/core/testdata/integration_serial_suite_in.json [hunk 1]
```diff
@@ -279,6 +279,7 @@
       "desc format = 'brief' select /*+hash_agg()*/ sum(id) from (select value, id from t where id > value group by id, value)A group by value /*the exchange should have only one partition column: test.t.value*/",
       "desc format = 'brief' select /*+hash_agg()*/ sum(B.value) from t as B where B.id+1 > (select count(*) from t where t.id= B.id and t.value=B.value) group by B.id /*the exchange should have only one partition column: test.t.id*/",
       "desc format = 'brief' select count(distinct value) from t",
+      "desc format = 'brief' select count(distinct x ) from (select count(distinct value) x from t) t",
```
Comment (fzhedu) [inline planner/core/testdata/integration_serial_suite_in.json:282]: would we add a test case : select count(distinct x), avg(y) ? becase avg would add a proj.

Comment (hanfei1991) [reply planner/core/testdata/integration_serial_suite_in.json:282]: the test case below this one is exactly what you mentioned

Comment (fzhedu) [reply planner/core/testdata/integration_serial_suite_in.json:282]: ok

**Review by fzhedu — APPROVED** [review]
/LGTM

**Review by winoros — APPROVED** [review]

**Pull request merged**